packages/keystone/src/fields/types/relationship/views/index.tsx

```diff
@@ -554,3 +662,41 @@ export const controller = (
     },
   };
 };
+
+function useRelationshipFilterValues({ value, list }: { value: string; list: ListMeta }) {
+  const foreignIds = getForeignIds(value);
+  const where = { id: { in: foreignIds } };
+
+  const query = gql`
+    query FOREIGNLIST_QUERY($where: ${list.gqlNames.whereInputName}!) {
+      items: ${list.gqlNames.listQueryName}(where: $where) {
+        id 
+        ${list.labelField}
+      }
+    }
+  `;
+
+  const { data, loading } = useQuery(query, {
+    variables: {
+      where,
+    },
+  });
+
+  return {
+    filterValues:
+      data?.items?.map((item: any) => {
+        return {
+          id: item.id,
+          label: item[list.labelField],
+        };
+      }) || foreignIds,
```

Comment:
`foreignIds` is an array of strings when it's expected that the returned thing is `{ id: string; label: string; }[]`

Reply:
Also, can you `item[list.labelField] || item.id` since the user might not have access to the label or something